### ISSUE TITLE
Breaks Nginx Stream Module

### ISSUE THREAD

**Issue by RohanNagar** [GUI/nginx-upstream-dynamic-servers#26]
When I compile this module with the nginx [stream module](http://nginx.org/en/docs/stream/ngx_stream_core_module.html), which is available since 1.9.0, the stream module breaks. Multiple stream tests are failing and resulting in seg faults.

I can try to fix this myself, but if anyone is still active I would appreciate some insight into why you think this might be happening.

I'm on version 1.13.5 of Nginx.

Thanks!

**Comment by wandenberg** [comment]
Hi @RohanNagar, I'm trying to fix some known bugs/segfaults in this module.
Can you explain the tests that you are doing? Or even better, provide a vagrant/docker/script where the bug can be easily reproduced?

**Comment by RohanNagar** [comment]
@wandenberg I am compiling Nginx 1.13.5 with this module, and the additional configure flags of

`--with-stream --with-stream_ssl_module`

I'm then running the core nginx tests found [here](https://github.com/nginx/nginx-tests). Almost all of the `stream` tests are failing.

Honestly, I'm not too sure what these tests are testing, I'm still new to Nginx. However it does seem like there are memory issues. Many times the tests expect a result and instead are getting `undefined` or empty strings. Also after many of the tests, the error "worker process exited with signal 11" are printed.

I can make a simple script to download these and run them if you would like.

**Comment by wandenberg** [comment]
Would be good to have a script.
Have you tried to run the tests without the module?

**Comment by RohanNagar** [comment]
Yes, the tests pass without the module. I will put a script together for you soon.
